test/jdk/java/nio/channels/FileChannel/Transfer.java

```diff
@@ -95,8 +95,11 @@ public void testFileChannel() throws Exception {
 
         bytesWritten = sinkChannel.transferFrom(sourceChannel, 1000, 10);
 
-        if (bytesWritten > 0)
-            throw new RuntimeException("Wrote past file size");
+        if (bytesWritten > 10)
+            throw new RuntimeException("Wrote too many bytes");
+
+        if (sinkChannel.size() != 1010)
```

Comment:
I assume this should be `1000 + bytesWritten` rather than 1010.

Reply:
Yes.

There are other places in here where it is not accounted for that transferX can return less than the count specified.

Reply:
Fixed in 6c138ef19704f57a34410e039fe5e0b3f5059bf3.